src/test/java/com/google/jenkins/plugins/computeengine/ConfigAsCodeTest.java

```diff
@@ -0,0 +1,53 @@
+package com.google.jenkins.plugins.computeengine;
+
+import static hudson.Functions.isWindows;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assume.assumeFalse;
+
+import com.cloudbees.plugins.credentials.Credentials;
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.SystemCredentialsProvider;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import com.google.jenkins.plugins.credentials.oauth.GoogleRobotPrivateKeyCredentials;
+import com.google.jenkins.plugins.credentials.oauth.ServiceAccountConfig;
+import io.jenkins.plugins.casc.misc.ConfiguredWithCode;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import org.apache.commons.io.IOUtils;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class ConfigAsCodeTest {
+
+  @Rule public JenkinsRule r = new JenkinsConfiguredWithCodeRule();
+
+  @Test
+  @ConfiguredWithCode("configuration-as-code.yml")
+  public void shouldCreateCloudInstanceFromCode() throws Exception {
+    assertEquals("Zero clouds found", r.jenkins.clouds.size(), 1);
+    ComputeEngineCloud cloud = (ComputeEngineCloud) r.jenkins.clouds.getByName("gce-jenkins-build");
```

Comment:
Would also like to see assertions verifying that the options specified in the config are being properly saturated.